code/modules/mob/living/carbon/life.dm

```diff
@@ -437,4 +438,22 @@
 				if(85 to INFINITY) severity = 6
 			overlay_fullscreen("brute", /obj/screen/fullscreen/brute, severity)
 		else
-			clear_fullscreen("brute")
\ No newline at end of file
+			clear_fullscreen("brute")
+
+/mob/living/carbon/proc/handle_patches()
+	if(LAZYLEN(processing_patches))
+		var/multiple_patch_multiplier = processing_patches.len > 1 ? (processing_patches.len * 1.5) : 1
+		var/applied_amount = 0.35 * multiple_patch_multiplier
+
+		for(var/patch in processing_patches)
+			var/obj/item/reagent_containers/food/pill/patch/P = patch
+
+			if(P.reagents && P.reagents.total_volume)
+				var/fractional_applied_amount = applied_amount  / P.reagents.total_volume
+				P.reagents.reaction(src, TOUCH, fractional_applied_amount, P.reagents.total_volume == P.reagents.maximum_volume)
```

Comment:
Wouldn't this mean that patches smaller than the maximum amount (less than 40) would not display a message?

Reply:
Yeah, it does. 

Any other ideas? `initial(P.reagents.total_volume) == P.reagents.total_volume`  doesn't work.

Reply:
Ideas:
- Set `maximum_volume` in initialize of a patch, although this could lead to problems if we somehow add more chems to a patch after creation
- Give a free "reaction" cycle directly on patch application, and don't show a message in the loop
- Add a flag to patches which says whether they've given their reaction message yet or not

Reply:
Been addressed!